### ISSUE TITLE
[bitnami/mysql:8.0.28-debian-10-r63] Access denied for user 'root'@'localhost' (using password: YES)

### ISSUE THREAD

**Issue by chargui-yassine** [bitnami/charts#9801]
### Name and Version

Bitnami MySQL Helm Chart Version 8.9.0 with image.tag: 8.0.28-debian-10-r63

### What steps will reproduce the bug?

`$ helm install mysql8 bitnami/mysql`

### Are you using any custom parameters or values?

No

### What is the expected behavior?

`ROOT_PASSWORD=$(kubectl get secret --namespace yassine-chargui mysql8 -o jsonpath="{.data.mysql-root-password}" | base64 --decode)`

```
k exec -it mysql8-0 -- mysql -u root -p$ROOT_PASSWORD
mysql>
```

### What do you see instead?

`ROOT_PASSWORD=$(kubectl get secret --namespace yassine-chargui mysql8 -o jsonpath="{.data.mysql-root-password}" | base64 --decode)`

```
k exec -it mysql8-0 -- mysql -u root -p$ROOT_PASSWORD
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
command terminated with exit code 1
```

### Additional information

```
`Name:         mysql8-0
Namespace:    mysql
Priority:     0
Node:         gke-devfactory-main-pool-e59263af-p28s/10.10.0.4
Start Time:   Thu, 14 Apr 2022 13:12:14 +0200
Labels:       app.kubernetes.io/component=primary
              app.kubernetes.io/instance=mysql8
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=mysql
              controller-revision-hash=mysql8-55bdddb7b
              helm.sh/chart=mysql-8.9.0
              statefulset.kubernetes.io/pod-name=mysql8-0
Annotations:  checksum/configuration: a5ba2bf54dfc75f85a229de8165041fc341b9a5509964bd5814bf68c5ab8c79a
              cni.projectcalico.org/containerID: 72f47d2e106733c2fe3be53b0b1dea0d5e0811d25f9e80e40188b7d1feed0390
              cni.projectcalico.org/podIP: 10.96.2.189/32
              cni.projectcalico.org/podIPs: 10.96.2.189/32
Status:       Running
IP:           10.96.2.189
IPs:
  IP:           10.96.2.189
Controlled By:  StatefulSet/mysql8
Containers:
  mysql:
    Container ID:   docker://a235cd8a1a336ec9e5ce6da46846b1db142d149c8fa713adfbed434efe6a5672
    Image:          docker.io/bitnami/mysql:8.0.28-debian-10-r63
    Image ID:       docker-pullable://bitnami/mysql@sha256:a73f8a78eb47cf06f82af4d5aa8a86056d8a7e18c69dee9967942a282a2e749b
    Port:           3306/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 14 Apr 2022 13:12:32 +0200
    Ready:          True
    Restart Count:  0
    Liveness:       exec [/bin/bash -ec password_aux="${MYSQL_ROOT_PASSWORD:-}"
if [[ -f "${MYSQL_ROOT_PASSWORD_FILE:-}" ]]; then
    password_aux=$(cat "$MYSQL_ROOT_PASSWORD_FILE")
fi
mysqladmin status -uroot -p"${password_aux}"
] delay=5s timeout=1s period=10s #success=1 #failure=3
    Readiness:  exec [/bin/bash -ec password_aux="${MYSQL_ROOT_PASSWORD:-}"
if [[ -f "${MYSQL_ROOT_PASSWORD_FILE:-}" ]]; then
    password_aux=$(cat "$MYSQL_ROOT_PASSWORD_FILE")
fi
mysqladmin status -uroot -p"${password_aux}"
] delay=5s timeout=1s period=10s #success=1 #failure=3
    Startup:  exec [/bin/bash -ec password_aux="${MYSQL_ROOT_PASSWORD:-}"
if [[ -f "${MYSQL_ROOT_PASSWORD_FILE:-}" ]]; then
    password_aux=$(cat "$MYSQL_ROOT_PASSWORD_FILE")
fi
mysqladmin status -uroot -p"${password_aux}"
] delay=15s timeout=1s period=10s #success=1 #failure=10
    Environment:
      BITNAMI_DEBUG:        false
      MYSQL_ROOT_PASSWORD:  <set to the key 'mysql-root-password' in secret 'mysql8'>  Optional: false
      MYSQL_DATABASE:       my_database
    Mounts:
      /bitnami/mysql from data (rw)
      /opt/bitnami/mysql/conf/my.cnf from config (rw,path="my.cnf")
      /var/run/secrets/kubernetes.io/serviceaccount from mysql8-token-qz9mt (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  data-mysql8-0
    ReadOnly:   false
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      mysql8
    Optional:  false
  mysql8-token-qz9mt:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  mysql8-token-qz9mt
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason                  Age   From                     Message
  ----     ------                  ----  ----                     -------
  Normal   Scheduled               51s   default-scheduler        Successfully assigned yassine-chargui/mysql8-0 to gke-devfactory-main-pool-e59263af-p28s
  Normal   SuccessfulAttachVolume  43s   attachdetach-controller  AttachVolume.Attach succeeded for volume "pvc-18e2cbec-48ab-4e9f-a30f-53ecfb6928ab"
  Normal   Pulled                  33s   kubelet                  Container image "docker.io/bitnami/mysql:8.0.28-debian-10-r63" already present on machine
  Normal   Created                 33s   kubelet                  Created container mysql
  Normal   Started                 33s   kubelet                  Started container mysql
  Warning  Unhealthy               9s    kubelet                  Liveness probe failed: mysqladmin: [Warning] Using a password on the command line interface can be insecure.
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mysql/tmp/mysql.sock' (2)'
Check that mysqld is running and that the socket: '/opt/bitnami/mysql/tmp/mysql.sock' exists!`
```

**Comment by Michiel-s** [comment]
I have the same issue. Tried to find out what is going wrong, but don't have a clue.

Below the startup logs I see for the pod:
```log
> kubectl logs db-production-mariadb-0

mariadb 15:24:06.02 
mariadb 15:24:06.02 Welcome to the Bitnami mariadb container
mariadb 15:24:06.02 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 15:24:06.02 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 15:24:06.02 
mariadb 15:24:06.02 INFO  ==> ** Starting MariaDB setup **
mariadb 15:24:06.03 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 15:24:06.04 INFO  ==> Initializing mariadb database
mariadb 15:24:06.04 DEBUG ==> Ensuring expected directories/files exist
mariadb 15:24:06.04 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment vari
ables will not be applied for this file.
mariadb 15:24:06.04 INFO  ==> Using persisted data
mariadb 15:24:06.06 INFO  ==> Running mysql_upgrade
mariadb 15:24:06.06 INFO  ==> Starting mariadb in background
2022-04-14 15:24:06 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.15-MariaDB) starting as process 47 ...
2022-04-14 15:24:06 0 [Note] InnoDB: Uses event mutexes
2022-04-14 15:24:06 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-04-14 15:24:06 0 [Note] InnoDB: Number of pools: 1
2022-04-14 15:24:06 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2022-04-14 15:24:06 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2022-04-14 15:24:06 0 [Note] InnoDB: Using Linux native AIO
2022-04-14 15:24:06 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2022-04-14 15:24:06 0 [Note] InnoDB: Completed initialization of buffer pool
2022-04-14 15:24:06 0 [Note] InnoDB: 128 rollback segments are active.
2022-04-14 15:24:06 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2022-04-14 15:24:06 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2022-04-14 15:24:06 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2022-04-14 15:24:06 0 [Note] InnoDB: 10.5.15 started; log sequence number 45130; transaction id 20
2022-04-14 15:24:06 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2022-04-14 15:24:06 0 [Note] InnoDB: Buffer pool(s) load completed at 220414 15:24:06
2022-04-14 15:24:06 0 [Note] Plugin 'FEEDBACK' is disabled.
2022-04-14 15:24:06 0 [Note] Server socket created on IP: '127.0.0.1'.
2022-04-14 15:24:06 0 [Warning] 'proxies_priv' entry '@% root@db-production-mariadb-0' ignored in --skip-name-resolve mode.
2022-04-14 15:24:06 0 [Note] Reading of all Master_info entries succeeded
2022-04-14 15:24:06 0 [Note] Added new Master_info '' to hash table
2022-04-14 15:24:06 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '10.5.15-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
2022-04-14 15:24:08 3 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
Reading datadir from the MariaDB server failed. Got the following error when executing the 'mysql' command line client
ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
FATAL ERROR: Upgrade failed
mariadb 15:24:08.08 INFO  ==> Stopping mariadb
2022-04-14 15:24:08 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
2022-04-14 15:24:08 0 [Note] Event Scheduler: Purging the queue. 0 events
2022-04-14 15:24:08 0 [Note] InnoDB: FTS optimize thread exiting.
2022-04-14 15:24:08 0 [Note] InnoDB: Starting shutdown...
2022-04-14 15:24:08 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
2022-04-14 15:24:08 0 [Note] InnoDB: Buffer pool(s) dump completed at 220414 15:24:08
2022-04-14 15:24:08 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2022-04-14 15:24:08 0 [Note] InnoDB: Shutdown completed; log sequence number 45142; transaction id 21
2022-04-14 15:24:08 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete

mariadb 15:24:09.09 DEBUG ==> Waiting for mariadb to unlock db files
```

**Comment by Michiel-s** [comment]
Ok, just before taking a brake now. I noticed that installing without `-f values.yaml` the issue is gone.

I'll try to find out why this is the case. Tbc.

**Comment by Michiel-s** [comment]
I think I have figured what the issue was for me, but not 100% sure. At least now it works for me.

What I did before installing with Helm, I used the following cmd to get the chart values in a yaml file
> `helm show values bitnami/mariadb --version 10.4.6 > db-values.yaml`

I'm working on a Windows machine with VS Code. The terminal I'm using is Powershell. The `db-values.yaml` file that is written has encoding `UTF-16 LE`. I didn't noticed that before.

After changing the encoding to `UFT-8` and then using that file to install with Helm, it worked:
> `helm install db-production bitnami/mariadb --version 10.4.6 -f db-values.yaml`

btw, I don't think this was the issue reported by @chargui-yassine as he didn't use custom values.
